plrust/src/user_crate/build.rs

```diff
@@ -180,15 +180,23 @@ impl FnBuild {
             let stdout = String::from_utf8(output.stdout).wrap_err("cargo stdout was not UTF-8")?;
             let stderr = String::from_utf8(output.stderr).wrap_err("cargo stderr was not UTF-8")?;
 
-            Err(eyre!(PlRustError::CargoBuildFail)
+            let err = Err(eyre!(PlRustError::CargoBuildFail)
                 .section(stdout.header("`cargo build` stdout:"))
                 .section(stderr.header("`cargo build` stderr:"))
                 .with_section(|| {
                     std::fs::read_to_string(&self.crate_dir.join("src").join("lib.rs"))
                         .wrap_err("Writing generated `lib.rs`")
                         .expect("Reading generated `lib.rs` to output during error")
                         .header("Source Code:")
-                }))?
+                }));
+
+            // Clean up on error
+            std::fs::remove_dir_all(&self.crate_dir).wrap_err(format!(
+                "Problem deleting temporary crate directory at '{}'",
+                self.crate_dir.display()
+            ))?;
```

Comment:
In the error case, if we fail to clean up, we should not fail to propagate the actual error message. This would do so. Attempt the cleanup without doing error propagation.

Reply:
Ahh yea that makes sense. I wasn't sure if it would happen to be problematic but it makes more sense to always throw the relevant compilation error